web/src/message_lists.ts

```diff
@@ -113,6 +114,7 @@ export function update_current_message_list(msg_list: MessageList | undefined):
     current = msg_list;
     if (current !== undefined) {
         rendered_message_lists.set(current.id, current);
+        message_list_data_cache.add(current.data);
```

Comment:
Hmm, what's the thinking about doing this here, rather than in `new MessageListData`? I guess doing it in the constructor has the risk that it'll try to scan itself when looking for candidates; so maybe this is the right way to do it.

Reply:
Yeah, also choosing this place would mean the data object is already populated with most data we have for it locally and will be updated if more data is added. This will mean it makes sense to use this data object instead of any we with the same filter in the cache too.